userspace/libscap/scap_userlist.c

```diff
@@ -76,11 +79,38 @@ int32_t scap_create_userlist(scap_t* handle)
 		return SCAP_FAILURE;		
 	}
 
+	// check for host root
+	const char *host_root = scap_get_host_root();
+	if(host_root[0] == '\0')
+	{
+		file_lookup = false;
+	}
+	else
+	{
+		file_lookup = true;
+	}
+
 	// users
-	setpwent();
-	p = getpwent();
+	if(file_lookup)
+	{
+		snprintf(filename, sizeof(filename), "%s/etc/passwd", host_root);
+		f = fopen(filename, "r");
+		if(f == NULL)
+		{
+			// if we don't have it inside the host root, we'll proceed without a list
+			free(handle->m_userlist->users);
+			free(handle->m_userlist->groups);
+			free(handle->m_userlist);
```

Comment:
And since I've already mentioned refactoring (and we really don't need to do this now, but it would be nice), we might want to either:
- Create a dedicated function for `free`'ing `m_userlist` and using that.
- Move code used for `free`'ing `m_userlist` to the end of the function and using a `goto` for cleanup.

Either option would reduce clutter and help prevent people making changes in the future from forgetting to free part of the struct.

Reply:
I didn't want to change the code much